sql/src/main/java/org/apache/druid/sql/calcite/expression/Expressions.java

```diff
@@ -214,12 +216,39 @@ public static DruidExpression toDruidExpressionWithPostAggOperands(
       return rexCallToDruidExpression(plannerContext, rowSignature, rexNode, postAggregatorVisitor);
     } else if (kind == SqlKind.LITERAL) {
       return literalToDruidExpression(plannerContext, rexNode);
+    } else if (kind == SqlKind.FIELD_ACCESS) {
+      return fieldAccessToDruidExpression(rowSignature, rexNode);
     } else {
       // Can't translate.
       return null;
     }
   }
 
+  private static DruidExpression fieldAccessToDruidExpression(
+      final RowSignature rowSignature,
+      final RexNode rexNode
+  )
+  {
+    // Translate field references.
+    final RexFieldAccess ref = (RexFieldAccess) rexNode;
+    // This case arises in the case of a correlation where the rexNode points to a table from the left subtree
+    // while the underlying datasource is the scan stub created from LogicalValuesRule
+    // In such a case we throw a CannotBuildQueryException so that Calcite does not go ahead with this path
+    // This exception is caught while returning false from isValidDruidQuery() method
+    if (ref.getField().getIndex() > rowSignature.size()) {
+      throw new CannotBuildQueryException(
+          "Cannot build query as index is higher than row size"
```

Comment:
suggest showing column/field name that is missing from row signature instead of saying anything about index since is confusing

Reply:
fixed